main/tests/server/src/com/google/refine/io/FileProjectManagerTests.java

```diff
@@ -116,4 +120,34 @@ public void deleteProjectAndSaveWorkspace() throws IOException {
         JsonObject json = JSON.parse(inputStream);
         assertTrue(json.get("projectIDs").getAsArray().isEmpty());
     }
+
+    /**
+     * Tests whether only meta files of modified projects will be updated locally.
+     */
+    @Test
+    public void metaFileUpdateTest() throws GetProjectIDException, InterruptedException {
+        FileProjectManager manager = new FileProjectManager(workspaceDir);
+        ProjectMetadata metaA = new ProjectMetadata();
+        ProjectMetadata metaB = new ProjectMetadata();
+        metaA.setName("A");
+        metaB.setName("B");
+        manager.registerProject(new Project(), metaA);
+        manager.registerProject(new Project(), metaB);
+        manager.saveWorkspace();
+        long idA = manager.getProjectID("A");
+        long idB = manager.getProjectID("B");
+        Path pathA = Paths.get(workspaceDir.getAbsolutePath(), String.valueOf(idA) + ".project", "metadata.json");
+        Path pathB = Paths.get(workspaceDir.getAbsolutePath(), String.valueOf(idB) + ".project", "metadata.json");
+        File metaAFile = pathA.toFile();
+        File metaBFile = pathB.toFile();
+        long timeBeforeA = metaAFile.lastModified();
+        long timeBeforeB = metaBFile.lastModified();
+        Thread.sleep(3000);
```

Comment:
This seems like a rather long sleep, I would expect we could at least halve it and the test would still pass reliably (even on systems where the lastModified() method has a second granularity - I do not actually know if this varies from system to system but I would expect all systems are at least precise to the second if not more).